App/dojo.py

```diff
@@ -50,142 +52,246 @@ def create_room(self, name, type_room):
+                return False
         else:
-            print("     ")
-            msg = 'ROOM %s NOT FOUND! WOULD YOU LIKE TO ADD?'% (room_name.upper())
-            click.secho(msg,bold=True, fg='red')
+            # if no room is created 
+            click.secho('Room %s seems not to exist would you like to add it?'\
+            % (room_name.capitalize()), fg='red', bold=True)
             print ("    ")
-                    # print (member.name)
-        
-    def print_unallocated(self, filename):
-        pass
+
     def print_allocations(self, filename):
-        pass
+        '''
+        This function prints a list of members(allocated) of a
+        specific room to the screen and to a file if opted
+        '''
+        # checks for rooms not added or created
+        rooms = self.all_rooms
+        if not rooms:
+            click.secho('No rooms found in the system',bold=True, fg='red')
+            return
+        output = ''
+        for room in rooms:
+            output += '==' * 20
+            output += '\n'
+            output += room.name.upper()
+            output += '\n'
+            output += '==' * 20
+            output += '\n'
+            # go through all members added
+            if room.members:
+                for member in room.members:
+                    output += member.name
+                    output += '\n'
+            else:
+                # prints out a created room with no members
+                output += 'There are no people  yet.'
+                output += '\n'
+        # This prints a list of all allocations onto the screen without the 
+        # optional --o
+        if filename == None:
+            click.secho(output, fg='yellow')
+
+        else:
+            # prints the list of allocations to a specified file(.txt)
+            click.secho('PLEASE WAIT...', bold=True, fg='magenta')
+            time.sleep(1)   
+            print(' ')          
+            text = ''
+            for room in rooms:
+                text += room.name.upper() + "\n"
+                text += "-" * 50 + "\n"
+                if room.members:
+                    text += " , " .join(member.name for member in room.members) + '\n'
+                    text += '\n'
+            # specify the fomart and mode        
+            file = open(filename + '.txt', 'w')
+            file.write(text)
+            click.secho('Your allocation list has been saved sucessfully \nas %s.txt' % filename,bold=True, fg='green')
+            return
+
+    def print_unallocated(self, filename):
+        allocated = self.all_people
+        self.unallocated = []
+        output = ''
+        output += "=" * 20 + "\n"
+        output += "Unallocated People\n"
+        output += "=" * 20 + "\n"
+        for person in allocated:
+            if person not in allocated:
+                output += person.name + '\n'
+                if person not in self.unallocated:
+                    self.unallocated.append(person)
```

Comment:
I think having the storage of unallocated person here is an unnecessary dependency. Assuming you need to get unallocated persons elsewhere, you will need to be sure that this method has run.